src/DelayBucket.cc

```diff
@@ -25,9 +27,19 @@ DelayBucket::stats(StoreEntry *entry)const
 void
 DelayBucket::update(DelaySpec const &rate, int incr)
 {
-    if (rate.restore_bps != -1 &&
-            (level() += rate.restore_bps * incr) > rate.max_bytes)
-        level() = rate.max_bytes;
+    if (rate.restore_bps == -1)
+        return;
+
+    const auto oldLevel = level();
+    const auto delta = IntegralProduct(oldLevel, rate.restore_bps, incr);
+    if (delta.has_value()) {
```

Comment:
```suggestion
    if (const auto delta = IntegralProduct(oldLevel, rate.restore_bps, incr)) {
```

Folks may misread this as protection against zero values, but I think it is still better to avoid "leaking" undefined `delta`.